test/CustomAccounting.t.sol

```diff
@@ -319,4 +332,59 @@ contract CustomAccountingTest is Test, Deployers, GasSnapshot {
         assertEq(thisGain0 + hookGain0, managerLoss0, "manager amount 0");
         assertEq(thisGain1 + hookGain1, managerLoss1, "manager amount 1");
     }
+
+    function test_fuzz_addLiquidity_withLPFeeTakingHook(uint128 feeRevenue0, uint128 feeRevenue1) public {
+        feeRevenue0 = uint128(bound(feeRevenue0, 0, type(uint128).max / 2));
+        feeRevenue1 = uint128(bound(feeRevenue1, 0, type(uint128).max / 2));
+        _setUpLPFeeTakingPool(); // creates liquidity as part of setup
+
+        // donate to generate fee revenue
+        donateRouter.donate(key, feeRevenue0, feeRevenue1, ZERO_BYTES);
+
+        uint256 hookBalanceBefore0 = currency0.balanceOf(hook);
+        uint256 hookBalanceBefore1 = currency1.balanceOf(hook);
+
+        // add liquidity again to trigger the hook, which should take the fee revenue
+        modifyLiquidityRouter.modifyLiquidity(key, LIQUIDITY_PARAMS, ZERO_BYTES);
+
+        uint256 hookGain0 = currency0.balanceOf(hook) - hookBalanceBefore0;
+        uint256 hookGain1 = currency1.balanceOf(hook) - hookBalanceBefore1;
+
+        // Assert that the hook took the fee revenue
+        assertApproxEqAbs(hookGain0, feeRevenue0, 1 wei);
+        assertApproxEqAbs(hookGain1, feeRevenue1, 1 wei);
```

Comment:
it irks me that addLiquidity assertions arent as robust as the removeLiquidity ones, _however_ this test is to confirm that `feesAccrued` is properly passed & used by the hook, so this is OK to me

---

behaviorally, the assertions for addLiquidity were tricky because of autocompounding + wei imprecision